### PR TITLE
meson

### PR DESCRIPTION
**Pull request by konradybcio**
depends on https://github.com/linux-msm/qrtr/pull/28

### CODE REVIEW

**Comment by superna9999** [comment]
Where do you change the .service.in file to fill the TQFTPSERVER_PATH variable ?

**Review by superna9999 — APPROVED** [review]
Patch order is weird, I would only enable the compiler args _after_ warnings fixups, otherwise it looks  good

File: tqftpserv.c [hunk 1]
```diff
@@ -81,7 +81,7 @@ static ssize_t tftp_send_data(struct tftp_client *client,
 	if (response_size != 0) {
 		/* Header (4 bytes) + data size */
 		send_len = 4 + response_size;
-		if (send_len > p - buf) {
+		if (send_len > (unsigned long long)(p - buf)) {
```
Comment (lumag) [inline tqftpserv.c:84]: This is called ptrdiff_t

File: tqftpserv.c [hunk 2]
```diff
@@ -138,7 +138,7 @@ static int tftp_send_oack(int sock, size_t *blocksize, size_t *tsize,
 		*p++ = '\0';
 	}
 
-	if (tsize && *tsize != -1) {
+	if (tsize && *tsize != -1U) {
```
Comment (lumag) [inline tqftpserv.c:141]: This is ridiculous IMO

Comment (konradybcio) [reply tqftpserv.c:141]: hey, I didn't come up with assigning -1 as an error value to a variable of type size_t /shrug

File: tqftpserv.c (hunk 1 above)
Comment (konradybcio) [reply tqftpserv.c:84]: ptrdiff_t is `long int`, so gcc still complains about signedness

Comment (lumag) [reply tqftpserv.c:84]: Then `size_t`.

**Comment by minlexx** [comment]
I've tried it and it seems to work; however compilation step gives me a lot of warnings: https://paste.sr.ht/~minlexx/4ef25af2db3d687961af9e1a9f9479c5db117c25

**Comment by konradybcio** [comment]
Dropped all the weird warnings, stepped down warnlevel to =1 (same as in qrtr), only kept the actual mesonification and spdx conversion @lumag @minlexx

**Review by lumag — APPROVED** [review]

**Pull request merged**